inst/templates/report_template.qmd

```diff
@@ -1,78 +1,91 @@
 ---
-title: "Performance comparison"
+title: "Performance report"
 format:
   html:
     toc: true
     toc_float: true
 params:
-  test_output: ""
+  performance: ""
 ---
 
 ```{r include = FALSE}
-suppressPackageStartupMessages({
-  library(DT)
-  library(dplyr)
-  library(echarts4r)
-})
+library(DT)
+library(dplyr)
+library(echarts4r)
+
 DT_options <- list(
   initComplete = JS(
     "function(settings, json) {",
     "$(this.api().table().header()).css({'background-color': '#1a384c', 'color': '#fff'});",
-    "}")
+    "}"
+  )
 )
 ```
 
-## Versions
+## App's Versions
 
-This report aims to compare different versions of the app. Here is the list of versions and the development dates:
+This report aims to compare different versions of your Shiny application. Here is the list of versions and the development dates (last commit for each version):
 
 ```{r data_prep, echo = FALSE}
-out <- params$test_output
-df_all_branches <- NULL
-for (n in range(1:length(out))) {
-  tmp_list <- out[[n]][[1]]
-  tmp_df <- data.frame(matrix(
-    unlist(tmp_list),
-    nrow = length(tmp_list$date),
-    byrow = FALSE
-  ))
-  colnames(tmp_df) <- names(tmp_list)
-  tmp_df$branch <- names(out)[n]
-  df_all_branches <- bind_rows(df_all_branches, tmp_df)
-}
-df_all_branches$date <-
-  format(as.POSIXct(
-    as.numeric(df_all_branches$date),
-    origin = "1970-01-01",
-    tz = "UTC"
-  ),
-  "%Y-%m-%d %H:%M:%S")
-df_all_branches$duration_ms <-
-  as.numeric(df_all_branches$duration_ms)
+# create an unique data.frame for all branches and repetitions
+df_all <- mapply(
+  params$performance,
+  names(params$performance),
+  FUN = function(x, y) {
+    df <- bind_rows(x)
+    df$branch <- y
+
+    return(df)
+  },
+  SIMPLIFY = FALSE
+)
+
+df_all <- bind_rows(df_all)
+df_all$duration_ms <- round(x = df_all$duration_ms, digits = 4)
```

Comment:
This throws an error:
```
processing file: report.qmd
  |................                                    |  31% [data_prep]      
Quitting from lines 30-60 [data_prep] (report.qmd)
Error in `round()`:
! non-numeric argument to mathematical function
                                                                                                            
Execution halted
```

Reply:
My guess is that benchmark is failing and then the output is empty. As a result, the report breaks since the data is not there. I added a new condition before running the report to prevent it from running in case benchmark did not run.